Src/DependencyCollector/NetCore/DependencyCollectorDiagnosticListener.cs

```diff
@@ -126,7 +124,7 @@ public void OnActivityStart(HttpRequestMessage request)
             if (request != null &&
                 !this.applicationInsightsUrlFilter.IsApplicationInsightsUrl(request.RequestUri.ToString()))
             {
-                this.InjectRequestHeaders(request, this.client.Context.InstrumentationKey);
+                this.InjectRequestHeaders(request, this.configuration.InstrumentationKey);
```

Comment:
I also discussed it with @nizarq and he thinks that for role correlation, it would be better to use some other source for the role information instead of telemetry properties